clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -100,6 +106,16 @@ protected boolean removeEldestEntry(Map.Entry<Schema, GenericRecord> eldest) {
 
   private static final ThreadLocal<ReusableObjects> REUSABLE_OBJECTS = ThreadLocal.withInitial(ReusableObjects::new);
 
+  /**
+   * The following two fields are used to speed up the requests with a big number of keys:
+   * 1. Split the big request into smaller chunks.
+   * 2. Execute these chunks concurrently.
+   */
+  private static final ExecutorService READ_CHUNK_EXECUTOR = Executors.newFixedThreadPool(
+      Runtime.getRuntime().availableProcessors(),
+      new DaemonThreadFactory("DaVinci_Read_Chunk_Executor"));
+  public static final int CHUNK_SPLIT_THRESHOLD = 100;
```

Comment:
probably can do larger threshold like 500, and it will be still under 10ms?

Reply:
Actually how about we use a config to control the chunk split threshold?

Reply:
I am trying to eliminate the config if possible.
Exposing config is not always a good practice as it might confuse customers.
100 chunk size should be a good threshold, as it would roughly take 2-3ms to execute one chunk.
10ms is typically not good enough for DVC users.

If this default won't work out, I will add a new config later.

Reply:
Minimizing config is always a good idea, but having an option to override has its benefit as well. Wondering if we make the default to 150 to match with thin-client and also eliminate unnecessary chunking for such cases.

Reply:
150 is mainly for minimizing Router retry problem and it is not really related to DaVinci.
Also this number is tightly related to the performance behavior of DaVinci.
I don't see a big difference between 100 and 150.

For DVC, the goal is as fast as possible instead of single-digit ms as all the data is in memory, and the trade-off is mainly the additional context switches and latency reduction.

If you folks insist it, I can add it to DaVinciConfig, but I guess not all the customers have the experience about how to tune it properly without consulting us.

Reply:
Point taken, and I will add it to DaVinciConfig to make it overridable.